### PR TITLE
switch to use kcorev1 informers in project cache

### PR DESCRIPTION
**Pull request by juanvallejo**
cc @deads2k

### CODE REVIEW

File: pkg/cmd/openshift-apiserver/openshiftapiserver/project.go [hunk 1]
```diff
@@ -14,17 +14,17 @@ import (
 	quotainformer "github.com/openshift/origin/pkg/quota/generated/informers/internalversion/quota/internalversion"
 )
 
-func NewClusterQuotaMappingController(nsInternalInformer kinternalinformers.NamespaceInformer, cqInternalInformer quotainformer.ClusterResourceQuotaInformer) *clusterquotamapping.ClusterQuotaMappingController {
-	return clusterquotamapping.NewClusterQuotaMappingControllerInternal(nsInternalInformer, cqInternalInformer)
```
Comment (deads2k) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/project.go]: this is now only used by a test right?  Fix the test.

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go [hunk 2]
```diff
@@ -246,7 +240,6 @@ func (i *KubeAPIServerInformers) GetOpenshiftUserInformers() userinformer.Shared
 }
 
 func (i *KubeAPIServerInformers) Start(stopCh <-chan struct{}) {
-	i.InternalKubernetesInformers.Start(stopCh)
```
Comment (deads2k) [inline pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: After this change, what is starting the informers?

File: pkg/cmd/openshift-apiserver/openshiftapiserver/project.go [hunk 3]
```diff
@@ -1,11 +1,11 @@
 package openshiftapiserver
 
 import (
+	kcoreinformers "k8s.io/client-go/informers/core/v1"
```
Comment (soltysh) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/project.go]: I think we should start keeping this consistent with our conventions, iow. this should be `corev1informers`

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go [hunk 4]
```diff
@@ -15,7 +15,6 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/kubernetes/cmd/kube-apiserver/app"
 	internalinformers "k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion"
-	kinternalinformers "k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion"
```
Comment (soltysh) [inline pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: I wonder how many places like this exist in the code :wink:

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go (hunk 2 above)
Comment (soltysh) [reply pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: Why would we care, if we're entirely dropping internal informers we don't need to. External informers are started in the next line, or I'm missing something.

File: pkg/cmd/openshift-apiserver/openshiftapiserver/project.go (hunk 3 above)
Comment (juanvallejo) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/project.go]: Makes sense. I have seen the `kcoreinformers` throughout the codebase, so I think it might make sense to sweep those in the future as well

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: internalInformers are still consumed within this file by the authorizer: https://github.com/openshift/origin/blob/master/pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go#L78

Since we are no longer starting internal kube informers in that method, I have updated `patchContext.informerStartFuncs` here https://github.com/openshift/origin/pull/21411/files#diff-01a8de689f656fdb0f202987ee6c1f2cR148

**Comment by juanvallejo** [comment]
@soltysh @deads2k thanks for the review, comments addressed

File: pkg/cmd/openshift-apiserver/openshiftapiserver/project.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/project.go]: Thanks, done

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go [hunk 5]
```diff
@@ -217,7 +215,6 @@ func NewInformers(internalInformers internalinformers.SharedInformerFactory, ver
 }
 
 type KubeAPIServerInformers struct {
-	InternalKubernetesInformers     kinternalinformers.SharedInformerFactory
```
Comment (deads2k) [inline pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: this feels good.

Comment (mfojtik) [reply pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: 👍

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go (hunk 4 above)
Comment (mfojtik) [reply pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: 28 :-)

File: pkg/project/util/util.go [hunk 6]
```diff
@@ -27,6 +29,16 @@ func ConvertNamespace(namespace *kapi.Namespace) *projectapi.Project {
...
 }
 
+// ConvertNamespace transforms a Namespace into a Project
+func ConvertNamespaceExternalOrDie(namespace *corev1.Namespace) *projectapi.Project {
+	internalNs := &kapi.Namespace{}
+	if err := kapiv1.Convert_v1_Namespace_To_core_Namespace(namespace, internalNs, nil); err != nil {
+		panic(err)
+	}
+
+	return ConvertNamespace(internalNs)
```
Comment (deads2k) [inline pkg/project/util/util.go]: don't do this.  rewrite the method for externals.  

Does anyone use the old one now?

Comment (juanvallejo) [reply pkg/project/util/util.go]: Thanks, removed call to other method.

@deads2k 
> Does anyone use the old one now?

It is still used by:
- https://github.com/openshift/origin/blob/master/pkg/project/apiserver/registry/project/proxy/proxy.go#L140
- https://github.com/openshift/origin/blob/master/pkg/project/apiserver/registry/project/proxy/proxy.go#L164
- https://github.com/openshift/origin/blob/master/pkg/project/apiserver/registry/project/proxy/proxy.go#L198
- https://github.com/openshift/origin/blob/master/pkg/project/auth/watch.go#L135
- https://github.com/openshift/origin/blob/master/pkg/project/auth/watch.go#L152

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go (hunk 4 above)
Comment (soltysh) [reply pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go]: @mfojtik oh that's the gist you were sending me earlier today :wink:

File: pkg/cmd/openshift-apiserver/openshiftapiserver/project.go [hunk 7]
```diff
@@ -1,11 +1,11 @@
 package openshiftapiserver
 
 import (
+	corev1informers "k8s.io/client-go/informers/core/v1"
 	rbacinformers "k8s.io/client-go/informers/rbac/v1"
+	kclientset "k8s.io/client-go/kubernetes"
```
Comment (soltysh) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/project.go]: We usually use this as `kubernetes`

File: pkg/project/auth/watch.go [hunk 8]
```diff
@@ -149,7 +149,7 @@ func (w *userProjectWatcher) GroupMembershipChanged(namespaceName string, users,
 
 		event := watch.Event{
 			Type:   watch.Added,
-			Object: projectutil.ConvertNamespace(namespace),
+			Object: projectutil.ConvertNamespaceExternalOrDie(namespace),
```
Comment (soltysh) [inline pkg/project/auth/watch.go]: I'm not sure we want to die here, @deads2k ?

File: pkg/cmd/openshift-apiserver/openshiftapiserver/project.go (hunk 7 above)
Comment (juanvallejo) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/project.go]: Thanks, updated import alias

File: pkg/project/auth/watch.go (hunk 8 above)
Comment (juanvallejo) [reply pkg/project/auth/watch.go]: I could switch this to return an error and use `utilruntime.HandleError` instead

File: pkg/project/util/util.go [hunk 9]
```diff
@@ -27,6 +29,24 @@ func ConvertNamespace(namespace *kapi.Namespace) *projectapi.Project {
 	}
 }
 
+// ConvertNamespace transforms a Namespace into a Project
+func ConvertNamespaceExternalOrDie(namespace *corev1.Namespace) *projectapi.Project {
+	internalNs := &kapi.Namespace{}
+	if err := kapiv1.Convert_v1_Namespace_To_core_Namespace(namespace, internalNs, nil); err != nil {
```
Comment (deads2k) [inline pkg/project/util/util.go]: don't convert to an internal ns and eliminate the possibility of an error.

Comment (juanvallejo) [reply pkg/project/util/util.go]: That would mean I'd have to return a `projectv1.Project` object. I am assuming that this would be okay, since the consumer of this function is a `watch.Event`?

Comment (juanvallejo) [reply pkg/project/util/util.go]: hm, this would mean that we would be returning v1 projects for some events, and non-versioned projects for others

File: pkg/project/apiserver/admission/nodeenv/admission_test.go [hunk 10]
```diff
@@ -25,6 +27,7 @@ func TestPodAdmission(t *testing.T) {
 	projectStore := projectcache.NewCacheStore(cache.IndexFuncToKeyFuncAdapter(cache.MetaNamespaceIndexFunc))
 	projectStore.Add(project)
 
+	externalMockClientset := fakev1.NewSimpleClientset()
 	mockClientset := fake.NewSimpleClientset()
```
Comment (deads2k) [inline pkg/project/apiserver/admission/nodeenv/admission_test.go]: this looks questionable.  What is it for?

File: pkg/project/auth/watch.go [hunk 11]
```diff
@@ -147,9 +147,14 @@ func (w *userProjectWatcher) GroupMembershipChanged(namespaceName string, users,
 			return
 		}
 
+		projectObj, err := projectutil.ConvertNamespaceExternal(namespace)
```
Comment (deads2k) [inline pkg/project/auth/watch.go]: this change looks wrong.  Why did you change the type?

File: pkg/project/util/util.go (hunk 9 above)
Comment (deads2k) [reply pkg/project/util/util.go]:
```go
internalFinalizers := rightTpe here
for _, externalFinalizer := externalNS.spec.Finalizers{
    internalFinalizers = append(internalFinalizers, internalFinalizer(string(externalFinalizer)))
}
return &projectapi.Project{
ObjectMeta: namespace.ObjectMeta,
		Spec: projectapi.ProjectSpec{
			Finalizers: internalFinalizers
		},
		Status: projectapi.ProjectStatus{
			Phase: internalNs.Status.Phase,
		},
}
```

No error

File: pkg/project/auth/watch.go (hunk 11 above)
Comment (deads2k) [reply pkg/project/auth/watch.go]: > this change looks wrong. Why did you change the type?

Ok I see.  See my comments on that method. No error, just build the right value.

**Comment by deads2k** [comment]
there is still an extraneous error and there is a questionable client in a test which suggests a type discrepancy in something underneath.

File: pkg/project/apiserver/admission/nodeenv/admission_test.go (hunk 10 above)
Comment (juanvallejo) [reply pkg/project/apiserver/admission/nodeenv/admission_test.go]: int the line below `handler := &podNodeEnvironment{client: ...}`, `podNodeEnvironment` requires an internal clientset

File: pkg/project/auth/watch.go (hunk 11 above)
Comment (juanvallejo) [reply pkg/project/auth/watch.go]: Thanks, addressed this using your suggestion

**Comment by juanvallejo** [comment]
@deads2k 
> there is still an extraneous error and there is a questionable client in a test which suggests a type discrepancy in something underneath.

Thanks, comments addressed
